executor/batch_point_get.go

```diff
@@ -421,6 +438,13 @@ func (e *BatchPointGetExec) initialize(ctx context.Context) error {
 		existKeys = make([]kv.Key, 0, 2*len(values))
 	}
 	e.values = make([][]byte, 0, len(values))
+	if e.lock && rc {
```

Comment:
This condition can be merged with L437-439.

https://github.com/pingcap/tidb/blob/cf4453df43a60b1fc5166521571343ef0cbcb686/executor/batch_point_get.go#L437-L439